scripts/pythondistdeps: Preparing PR for upstream

torsava opened the pull request:
First two commits are upstreaming of changes from Fedora.

torsava committed:
scripts/pythondistdeps: Add option to generate major-version provides only for specified Python versions

torsava committed:
scripts/pythondistdeps: Also provide pythonXdist() with PEP 503 normalized names

Upstreaming from Fedora (BZ#1791530)

That is, we add new provides that replace dots with a dash.

Package that used to provide python3dist(zope.component) and python3.8dist(zope.component)
now also provides python3dist(zope-component) and python3.8dist(zope-component).

Package that used to provide python3dist(a.-.-.-.a) now provides python3dist(a-a) as well.

This is consistent with pip behavior, `pip install zope-component` installs zope.component.

Historically, we have always used dist.key (safe_name) from setuptools,
but that is a non-standardized convention -- whether or not it replaces dots
with dashes is not even documented.
We say we use "canonical name" or "normalized name" everywhere, yet we didn't.

We really need to follow the standard (PEP 503):

https://www.python.org/dev/peps/pep-0503/#normalized-names

The proper function here would be packaging.utils.canonicalize_name
https://packaging.pypa.io/en/latest/utils/#packaging.utils.canonicalize_name
-- we reimplement it here to avoid an external dependency.

This is the first required step needed if we want to change our requirements later.
If we decide we don't, for whatever reason, this doesn't break anything.

torsava committed:
scripts/pythondistdeps: "Fix" support of environment markers

Upstreaming from Fedora

torsava committed:
scripts/pythondistdeps: Implement --normalized-name-* options

--normalized-names-format FORMAT
    FORMAT of normalized names can be `pep503` [default] or `legacy-dots` (dots allowed)

--normalized-names-provide-both
    Provede both `pep503` and `legacy-dots` format of normalized names (useful for a transition period)

torsava committed:
scripts/pythondistdeps: Add tests

The test data download themselves using pip if not present

torsava commented:
@hroncok if you want to take a look at what I have right now. I'm going to sleep :)

hroncok commented:
Note to the future: Maintaining the test data and tests in upstream RPM repository and/or in downstream dist-git is going to be painful. For now, that's good, but in the future this needs to be solved somehow.

torsava commented:
TBD: Move tests to some permanent location, ideally hook them into the existing test structure (if possible).

torsava commented:
> Note to the future: Maintaining the test data and tests in upstream RPM repository and/or in downstream dist-git is going to be painful. For now, that's good, but in the future this needs to be solved somehow.

The tests I think are fine, do you see any concrete issues with having them alongside the code in the git repo? The test data I agree, dunno how to approach it better right now.

torsava committed:
scripts/pythondistdeps: Notes from an attempted rewrite to importlib.metadata

Notes from an attempted rewrite from pkg_resources to importlib.metadata in 2020:
1. While pkg_resources can open a metadata on a specified path
   (Distribution.from_location()), importlib provides access only to
   "installed package metadata", i.e. the the dist-info or egg-info directory
   must be "discoverable", i.e. on the sys.path.
   - Thankfully only the dist/egg-info directory must exist, the
     corresponding Python module does not have to be present.
   - The problems this causes:
     (a) You have to manipulate the sys.path to add the specific location of
         the site-packages directory inside the buildroot
     (b) If you have package "foo" in this newly added directory on sys.path
         and there is some problem and its dist/egg-info metadata are not found,
         importlib.metadata continues searching the sys.path and may discover a
         package with the same name (possibly same version) outside the
         buildroot.
         To get around this, you can manipulate the sys.path to remove all
         other "site-packages" directories. But you have to leave the
         standard library there, because importlib may import other modules
         (in my testing: base64, quopri, random, socket, calendar, uu)
     (c) I have not tested how well it works if you're ispecting metadata of
         different Python versions than the one you run the script with
         (especially Python 2 vs Python 3). This might also cause problems with
         dependency specifiers (i.e. python_version != "3.4")
2. Handling of dependencies (requires) is problematic in importlib.metadata
   - pkg_resources provides a way to separately list standard requires and a
     requires for each "extras" category. importlib does not provide this, it
     only spits out a list of strings, each string in the format:
     - 'packaging>=14',
     - 'towncrier>=18.5.0; extra == "docs"', or
     - 'psutil<6,>=5.6.1; (python_version != "3.4") and extra == "testing"
     you can either parse these with a regex (fragile) or use the external
     `packaging` Python module. `packaging`, however, also doesn't have a great
     support for figuring out extra dependencies, it provides the marker api:
     - <Marker(\'python_version != "3.4" and extra == "testing"\')>
     you can use Marker api to evaluate the condition, but not to parse.
     For parsing you can access the private api Marker._markers:
     - marker._markers=[[(<Variable('python_version')>, <Op('!=')>, \
           <Value('3.4')>)], 'and', (<Variable('extra')>, <Op('==')>, \
           <Value('testing')>)]
     which beyond the problem of being private is also not very useful for
     parsing due to its structure.
   - pkg_resources also provides version parsing, which importlib does not
     and `packaging` needs to be used
   - importlib is part of the standard library, but packaging and its
     2 runtime dependencies (pyparsing and six) are not, and therefore we
     would go from 1 dependency to 3
3. A few minor issues, more in the next section about equivalents.

importlib.metadata.distribution equivalents of pkg_resources.Distribution attributes:
- pkg_resources: dist.py_version
  importlib: # not implemented (but can be guessed from the /usr/lib/pythonXX.YY/ path)
- pkg_resources: dist.project_name
  importlib: dist.metadata['name']
- pkg_resources: dist.key
  importlib: # not implemented
- pkg_resources: dist.version
  importlib: dist.version
- pkg_resources: dist.requires()
  importlib: dist.requires  # but returns strings with almost no parsing done, and also lists extras
- pkg_resources: dist.requires(extras=dist.extras)
  importlib: # not implemented, has to be parsed from dist.requires
- pkg_resources: dist.get_entry_map('console_scripts')
  importlib: [ep for ep in importlib.metadata.entry_points()['console_scripts'] if ep.name == pkg][0]
             # I have not found a better way to get the console_scripts
- pkg_resources: dist.get_entry_map('gui_scripts')
  importlib: # Presumably same as console_scripts, but untested

torsava commented on scripts/pythondistdeps.py:
```diff
@@ -272,7 +344,78 @@
+#      only spits out a list of strings, each string in the format:
+#      - 'packaging>=14',
+#      - 'towncrier>=18.5.0; extra == "docs"', or
+#      - 'psutil<6,>=5.6.1; (python_version != "3.4") and extra == "testing"
+#      you can either parse these with a regex (fragile) or use the external
+#      `packaging` Python module. `packaging`, however, also doesn't have a great
+#      support for figuring out extra dependencies, it provides the marker api:
+#      - <Marker(\'python_version != "3.4" and extra == "testing"\')>
+#      you can use Marker api to evaluate the condition, but not to parse.
+#      For parsing you can access the private api Marker._markers:
+#      - marker._markers=[[(<Variable('python_version')>, <Op('!=')>, \
+#            <Value('3.4')>)], 'and', (<Variable('extra')>, <Op('==')>, \
+#            <Value('testing')>)]
+#      which beyond the problem of being private is also not very useful for
+#      parsing due to its structure.
+#    - pkg_resources also provides version parsing, which importlib does not
+#      and `packaging` needs to be used
+#    - importlib is part of the standard library, but packaging and its
+#      2 runtime dependencies (pyparsing and six) are not, and therefore we
+#      would go from 1 dependency to 3
+# 3. A few minor issues, more in the next section about equivalents.
+#
+# importlib.metadata.distribution equivalents of pkg_resources.Distribution attributes:
+# - pkg_resources: dist.py_version
+#   importlib: # not implemented (but can be guessed from the /usr/lib/pythonXX.YY/ path)
+# - pkg_resources: dist.project_name
+#   importlib: dist.metadata['name']
+# - pkg_resources: dist.key
+#   importlib: # not implemented
+# - pkg_resources: dist.version
+#   importlib: dist.version
+# - pkg_resources: dist.requires()
+#   importlib: dist.requires  # but returns strings with almost no parsing done, and also lists extras
+# - pkg_resources: dist.requires(extras=dist.extras)
+#   importlib: # not implemented, has to be parsed from dist.requires
+# - pkg_resources: dist.get_entry_map('console_scripts')
+#   importlib: [ep for ep in importlib.metadata.entry_points()['console_scripts'] if ep.name == pkg][0]
+#              # I have not found a better way to get the console_scripts
+# - pkg_resources: dist.get_entry_map('gui_scripts')
+#   importlib: # Presumably same as console_scripts, but untested
```
@hroncok What do you think?

hroncok replied:
The content looks reasonable but I wouldn't put such a long comment to the actual code. Not sure where to put it, but I'd simply comment this to the final PR, explaining why we have not ported it to importlib metadata. WDYT?

torsava replied:
I think it should be somewhere discoverable for when somebody else has the same idea 2 years from now. I agree the current location is not great.

How aobut adding a comment to the code: "# Notes from an attempted rewrite from pkg_resources to importlib.metadata in 2020 can be found in the commit message that added this code line"? and putting it in the commit message?

hroncok replied:
Add a short note to pkg_reosurces import saying that a port to importlib.resources was attempted in 2020 and failed, see commit message for details ?

torsava replied:
Exactly. Ok, doing it.

torsava commented on scripts/pythondistdeps.py:
```diff
@@ -10,27 +10,40 @@
 # RPM python dependency generator, using .egg-info/.egg-link/.dist-info data
 #
 
+# Please know:
+# - Notes from an attempted rewrite from pkg_resources to importlib.metadata in
+#   2020 can be found in the message of the commit that added this line.
+
```
@hroncok Changed to this.

torsava commented on scripts/pythondistdeps.py:
```diff
@@ -102,6 +139,9 @@
             lower.endswith('.egg-info') or \
             lower.endswith('.dist-info'):
         # This import is very slow, so only do it if needed
+        # - Notes from an attempted rewrite from pkg_resources to
+        #   importlib.metadata in 2020 can be found in the message of
+        #   the commit that added this line.
```
@hroncok And this.

hroncok commented:
See also `https://github.com/rpm-software-management/rpm/pull/1015`.

torsava commented:
> See also `https://github.com/rpm-software-management/rpm/pull/1015`.

Dayum, that's exactly what I was about to dig deep into. Well, good then :D I'll try to run it against the tests I prepared.

hroncok commented:
> exactly what I was about to dig deep into

Glad I have saved the day! Was scared it will come too late.

torsava commented:
> Glad I have saved the day! Was scared it will come too late.

I did some preliminary work, but nothing too big. I quickly realized I would have to drastically rewrite it like the gentleman in the PR, and decided to do the importlib first :) Thanks for alerting me!

torsava commented:
Added the commit from this PR to properly test it: https://github.com/rpm-software-management/rpm/pull/1015

Added tests for version handling.

torsava commented:
Note that currently the metadata for testing are saved in git. The plan is to remove them, and create a pytest fixture to download the metadata from PyPI before running the tests. Except for the `pyreq2rpm.tests` metadata which I've concocted myself.

hroncok commented:
@torsava You've asked for some code similar to the planed pytest fixture. I only know about this: https://github.com/pradyunsg/vendoring which originated in https://github.com/pypa/pip/blob/19.0.3/tasks/vendoring/__init__.py (that code is much shorter).

It does something else entirely, but might have some good ideas.

torsava commented:
Rebased on top of the version handling PR, all tests are passing.

torsava commented:
Rebased on master after the version handling got merged.

torsava commented:
There's again been a *lot* of activity, so I'l doing a rather time-consuming rebase.

torsava commented:
I caught the exception and raised a new one from it, the output looks pretty good, @hroncok 


    Traceback (most recent call last):
      File "/home/torsava/git/upstream/rpm/scripts/pythondistdeps.py", line 140, in convert
        return OPERATORS[operator](name, operator, version_id)
      File "/home/torsava/git/upstream/rpm/scripts/pythondistdeps.py", line 115, in convert_ordered
        version_id = version_id[40]
    IndexError: string index out of range
    
    The above exception was the direct cause of the following exception:
    
    Traceback (most recent call last):
      File "/home/torsava/git/upstream/rpm/scripts/pythondistdeps.py", line 401, in <module>
        spec_list.append(convert(name, spec[0], spec[1]))
      File "/home/torsava/git/upstream/rpm/scripts/pythondistdeps.py", line 142, in convert
        raise RuntimeError("Cannot process Python package version: " + version_id) from exc
    RuntimeError: Cannot process Python package version `2.4.8.*` for name `python3.9dist(foobar47)`

torsava committed:
scripts/pythondistdeps: Do anything only when called as a main script

Note that the code is completely unchanged except for the indentation
under the new if __name__ == "__main__":

Note that this change is necessary, but not sufficient to use the
RpmVersion class.
The init of the RpmVersion class will fail when called from an outside
script, because the `parse_version()` function is lazily imported from
the code outside the class.  However, adding the import of
parse_version() to RpmVersion class is not done right now, because while
we would import it from `pkg_resources`, other scripts might want to
rely instead of the lightweight `packaging` module for the import. Thus
I'm leaving this conondrum to be addressed in the future.

hroncok commented:
That's exactly how I would have done it. Keep rocking :P

torsava committed:
scripts/pythondistdeps: Version handling exception with better information

torsava committed:
scripts/pythondistdeps: Add the tests to the readme and makefile

torsava commented:
@hroncok Ready for review!

torsava commented:
I highly recommend reviewing commit-by-commit instead of by files changed.

hroncok commented on .gitignore:
```diff
@@ -81,3 +81,4 @@ Makefile.in.in
 /tests/rpmtests.dir
 /tests/rpmtests.log
 /tests/testing/
+tests/data/scripts_pythondistdeps/usr/
```
For consistency.

```suggestion
/tests/data/scripts_pythondistdeps/usr/
```

torsava replied:
Oh thank you, that would have bothered me if I realized it :D

hroncok commented on tests/README:
```diff
@@ -34,3 +34,12 @@ See
 the test suite unless a very restricted set of test is run.
+
+
+To test the pythondistdeps.py script, use
```
What dependencies do i need? pytest, pyyaml and pip?

torsava replied:
Plus setuptools which is a dependency of pythondistdeps themselves, added.

hroncok commented on tests/test_scripts_pythondistdeps.py:
```diff
@@ -0,0 +1,199 @@
+#
+# Test cases are saved in test-data.yaml inside TEST_DATA_PATH
+#   (currently ./data/scripts_pythondistdeps/)
+# - To regenerate this yaml file with the current results of pythondistdeps for
```
*this* yaml file is what?

hroncok commented on tests/data/scripts_pythondistdeps/test-requires.yaml:
```diff
@@ -0,0 +1,91 @@
+setuptools:
+    wheel:
+        41.6.0: [2.7, 3.7, 3.9]
```
Is it possible to have some tests for 3.10 as well to test double digits?

torsava replied:
Actually it is, for "sdist" I'm making the version up as I like. Ok, why not.

torsava replied:
And I found a problem in the test-data getting script :laughing:

torsava replied:
Ugh, yaml takes `3.10` as a float and thus returns `3.1`. I guess I'll have to quote all the values.

hroncok reviewed (commented):
I couldn't resist and I've skimmed over the overall diff. The script diff is indeed unreadable due to indentation changes. Will review individual commit tmrw. Here are some nitpicks, feel free to ignore them before the more thorough review.

hroncok commented on tests/test_scripts_pythondistdeps.py:
```diff
@@ -0,0 +1,199 @@
+def run_pythondistdeps(provides_params, requires_params, dist_egg_info_path):
+    """Runs pythondistdeps.py on `dits_egg_info_path` with given
+    provides/requires parameters and returns a tuple of generated
+    (provides, requires)"""
```
returns a dict

torsava replied:
+1

torsava commented:
Force pushed with fixes for all of those, thanks for noticing!

torsava commented on tests/test_scripts_pythondistdeps.py:
```diff
@@ -0,0 +1,198 @@
+                            try:
+                                # run_module() alters sys.modules and sys.argv, but restores them at exit
+                                runpy.run_module("pip", run_name="__main__", alter_sys=True)
+                            except SystemExit as exc:
```
@hroncok I misread it when we were screen sharing. It's "SystemExit", i.e. it's fired when pip finishes successfully or not!

If we remove the except completely, the script ends after the one launch of pip.

And I already have an error message built in later, so doing a pass here even when pip fails seems to me ok. Output when there's a pip failure:

    ERROR: unknown command "installx" - maybe you meant "install"
    Problem occured while getting dist-info/egg-infofor package 'setuptools', version '41.6.0'for Python 2.7

Thus I think we should leave it as is.

hroncok replied:
Right. Thanks.

torsava commented:
@hroncok Ready for review. Please take a look at all the fixup commits (some are added in the middle of the commit list).

And I've split the test-data into the relevant commits for easier readability.

Also, the trailing blank line in the commit is gone :D

hroncok commented on scripts/pythondistdeps.py:
```diff
@@ -247,10 +247,15 @@ def normalize_name(name):
-            # Python version while evaluating packages for different versions,
-            # therefore we mock the platform.python_version function.
-            # Discussed upstream: https://github.com/pypa/setuptools/pull/1275
+            # dependency is relevant based on environment markers [0],
```
```suggestion
            # dependency is relevant based on environment markers [1],
```

torsava replied:
Fixed manually

hroncok commented:
> And I've split the test-data into the relevant commits for easier readability.

I see the major versions split, but not the normalized one.

Never mind. Please, do rebase the fixups, so I can do a final check. Thanks.

hroncok commented on tests/data/scripts_pythondistdeps/test-requires.yaml:
```diff
@@ -0,0 +1,93 @@
+        '1.12.0': ['2.7', '3.7', '3.9']
+tox:
+    wheel:
+        '3.13.2': ['2.7', '3.7', '3.9']
```
tox 3.14 has:

```
install_requires =
    importlib-metadata >= 0.12, <1;python_version<"3.8"
```

I would like to see it tested: that importlib-metadata is required on 3.7 but not 3.8. Unfortunately 3.13.2 doesn't yet do this.

Or is there some other package here that does this?

torsava replied:
I've switched to `tox 3.14`, and also `kubernetes` and `astroids` have these conditionals. I checked the outputs, they look good.

torsava commented:
Unrelated to this PR, but the version `11.0.0b2` is converted into `11~b2`, I gotta check if that's correct.

EDIT: according to [pyreq2rpm](https://github.com/gordonmessmer/pyreq2rpm/blob/master/tests/test_convert.py) it is correct:

        (['foobar', '==', '2.0.0b5'], 'foobar = 2~b5'),

And a cursory reading of PEP 440 looks like it confirms it, so all is well.

torsava commented:
> I see the major versions split, but not the normalized one.
> 
> Never mind. Please, do rebase the fixups, so I can do a final check. Thanks.

They were in the fixup commit. Sorry, yes, there's too many commits :) I'll squash all and you'll see it more clearly.

hroncok commented:
> version `11.0.0b2` is converted into `11~b2`

yes, that is in fact correct.

The only problem is that `11~devX` sorts higher than `11~bX`. Hence an exception must be made to to convert to `11~~devX`. Luckily alpha and beta sort fine.

```
$ rpmdev-vercmp 11~b2 11~dev1
11~b2 < 11~dev1
$ rpmdev-vercmp 11~b2 11~0.dev1
11~b2 < 11~0.dev1
$ rpmdev-vercmp 11~b2 11~~dev1
11~b2 > 11~~dev1
```

Do you have a dev version package? See https://pypi.org/project/taskotron-python-versions/

torsava commented:
> Do you have a dev version package? See https://pypi.org/project/taskotron-python-versions/

What do you mean?

The package I found this in is `kubernetes==11.0.0b2`.

torsava commented:
@hroncok All squashed and beautiful, take a look!

hroncok commented:
>> Do you have a dev version package? See https://pypi.org/project/taskotron-python-versions/
>
> What do you mean?

Have a test with a `dev` versioned package as well. Not just `b`.

hroncok commented:
Commits look good! All squashed and beautiful. Thanks.

torsava commented:
> Have a test with a `dev` versioned package as well. Not just `b`.

Indeed I don't, and there's no such test on pyreq2rpm. Do you have a package that you know of? If not, I'll dig for some.

hroncok commented:
Yes, see https://pypi.org/project/taskotron-python-versions/

hroncok commented:
(Handle the dev test and fix in an extra commit please, it might change the behavior.)

torsava commented:
Well, there's some deeper problem with developer versions :)

    $ find tests/data/scripts_pythondistdeps/usr/lib/python3.9/site-packages/taskotron-python-versions-0.1.dev6.dist-info | python scripts/pythondistdeps.py --provides
    python3.9dist(taskotron) = python

Wtf

hroncok commented:
Wtf indeed.

torsava commented:
Well, I'm getting it already as input from pkg_resources, so it goes even deeper.

hroncok commented:
maybe taskotron-python-versions are broken?

hroncok commented:
where is taskotron-python-versions-0.1.dev6.dist-info from?

torsava commented:
> where is taskotron-python-versions-0.1.dev6.dist-info from?

Discussed off-github, we found that the test script was not saving the dist-info with the proper directory name, it should have been taskotron_python_versions_*. Fixed.

torsava committed:
scripts/pythondistdeps: Modify handling of dev versions

torsava commented:
> The only problem is that `11~devX` sorts higher than `11~bX`. Hence an exception must be made to to convert to `11~~devX`. Luckily alpha and beta sort fine.

Done, PTAL @hroncok

torsava commented:
We might also want to propose the `~~dev` handling to pyreq2rpm.

hroncok commented on tests/test_scripts_pythondistdeps.py:
```diff
@@ -0,0 +1,238 @@
+                            try:
+                                info_path = next(temp_path.glob(f"**/*{suffix}"))
+
+                                # pro wheel pojdme assertovat ze info path se rovna install path
```
English? :D

torsava replied:
Omg, that was just my internal note from our review, not something I wanted to keep. Oops :D

hroncok commented on tests/data/scripts_pythondistdeps/pyreq2rpm.tests-2020.04.07.024dab0-py3.9.egg-info/requires.txt:
```diff
@@ -0,0 +1,95 @@
+foobar29===2.4.8b5
+foobar30===2.0.0b5
+foobar31===2.4.8.post1
+foobar32===2.0.post1
```
Add dev versuon requirement somewhere here to also test requires, not only provides.

torsava replied:
+1

hroncok commented:
The convert function is unusable because parse_version is not imported. We might want to reconsider the whole "make it importable" approach later and fix this. At least acknowledge in the commit message that this is a necessary step to make it reusable, but we are not there yet.

torsava commented:
> The convert function is unusable because parse_version is not imported. We might want to reconsider the whole "make it importable" approach later and fix this. At least acknowledge in the commit message that this is a necessary step to make it reusable, but we are not there yet.

Why not just add `from pkg_resources import parse_version` inside `RpmVersion.__init__`? It's already imported at that point, so I don't think it's going to cause a slowdown.

torsava commented:
> Why not just add `from pkg_resources import parse_version` inside `RpmVersion.__init__`? It's already imported at that point, so I don't think it's going to cause a slowdown.

Just ran it on the test data, and there's no slow down at all.

hroncok commented:
> Why not just add from pkg_resources import parse_version inside RpmVersion.__init__? It's already imported at that point, so I don't think it's going to cause a slowdown.

Please don't. I'd tell you why but I only have like 5 seconds before another meeting.

I'll send a PR to https://github.com/gordonmessmer/pyreq2rpm/blob/master/pyreq2rpm/pyreq2rpm.py with changes I deem useful and they can be backported to the genarators. Don't hold this PR upstream on it.

torsava commented:
> Please don't. I'd tell you why but I only have like 5 seconds before another meeting.

Ok, ping me when you have time. I still have to document it why it's not importable, so I need to understand it before opening an upstream PR.

>I'll send a PR to https://github.com/gordonmessmer/pyreq2rpm/blob/master/pyreq2rpm/pyreq2rpm.py with changes I > deem useful and they can be backported to the genarators. Don't hold this PR upstream on it.

I just tested that the `~~dev` works properly and it conforms to the [PEP ordering](https://www.python.org/dev/peps/pep-0440/#summary-of-permitted-suffixes-and-relative-ordering), so I was about to start on the PR. Do you want to make some other changes as well?

hroncok commented:
> I was about to start on the PR

Please do, my changes will follow on top of that.

torsava commented:
@hroncok Added some more tests. Ready for review.

torsava commented:
Fixups squashed, tests passing. How about it, @hroncok ?

torsava commented:
PR for pyreq2rpm: https://github.com/gordonmessmer/pyreq2rpm/pull/6

hroncok commented:
Go upstream. Thank you!

hroncok reviewed (approved):
![useless gif](https://www.picgifs.com/glitter-gifs/y/you-rock/picgifs-you-rock-1040010.gif)

torsava commented:
@hroncok Thanks for all the fish!

torsava commented:
Upstream PR: https://github.com/rpm-software-management/rpm/pull/1195

hroncok commented:
Close this to avoid double notifications?

torsava committed:
fixup scripts/pythondistdeps: Add the tests to the readme and makefile

Add test files to EXTRA_DIST in the Makefile.am

torsava commented:
Closing this preparatory PR, the real PR is already open.
